tests/unit_tests/arelle/test_pluginmanager.py

```diff
@@ -73,3 +80,98 @@ def test_plugin_manager_reset():
     assert len(PluginManager.modulePluginInfos) == 0
     assert len(PluginManager.pluginMethodsForClasses) == 0
     assert PluginManager._cntlr == cntlr
+
+@pytest.mark.parametrize(
+    "test_data, expected_result",
+    [
+        # Test case 1
+        (
+            # Test data
+            ("tests/unit_tests/arelle", "functionsMaths", "xyz"),
+            # Expected result
+            ("functionsMaths", "tests/unit_tests", "xyz")
+        ),
+        # Test case 2
+        (
+            # Test data
+            ("arelle/plugin/", "sphinx/__init__.py", "xyz"),
+            # Expected result
+            ("sphinx", "arelle/plugin", "sphinx.")
+        ),
+        # Test case 3
+        (
+            # Test data
+            ("plugin/sphinx", None, "xyz"),
+            # Expected result
+            (None, None, None)
+        ),
+    ]
+)
+def test_function_get_name_dir_prefix(
+    test_data: tuple[str, str, str],
+    expected_result: tuple[str, str, str],
+    ):
+    """Test util function get_name_dir_prefix."""
+    class Controller(Cntlr):
+        """Controller."""
+
+        pluginDir = test_data[0]
+
+        def __init__(self) -> None:
+            """Init controller with logging."""
+            super().__init__(logFileName="logToPrint")
+
+    cntlr = Controller()
+    PluginManager.init(cntlr, loadPluginConfig=False)
```

Comment:
This can be removed. `PluginManager.init` is already called from `super().__init__` in your test Controller